pkg/xds/generator/outbound_proxy_generator.go

```diff
@@ -188,10 +188,13 @@ func (_ OutboundProxyGenerator) determineSubsets(proxy *model.Proxy, outbound *k
 			// 0 assumes no traffic is passed there. Envoy doesn't support 0 weight, so instead of passing it to Envoy we just skip such cluster.
 			continue
 		}
+		// we need Mesh to be set as a tag and at the end be compiled into SNI. When Dataplane is talking with Ingress
+		// it has to provide own mesh in order to have an access only to the Dataplanes of the same mesh (in another zone)
+		withMesh := envoy_common.Tags(destination.Destination).WithTags("mesh", proxy.Dataplane.GetMeta().GetMesh())
```

Comment:
Ok so what I don't like about this is that it will trigger subset loadbalancing mechanism in Envoy every time, even if we don't use it really, because all endpoints are of the same mesh. I'm not familiar with performance of subset loadbalancing but I'd avoid it if it's not needed.

Extra tag in ingress will generate more permutations
```
{service=backend}
{mesh=default,service=backend}
```
but it does not make sense to generate permutations without mesh, because we will always require mesh from now one.

In my opinion it's better to not include `mesh` as a tag. In Ingress, include it as a separate field. Add `mesh` in SNI but not based on the tag.